sqlglot/dialects/dialect.py

```diff
@@ -347,7 +347,7 @@ def var_map_sql(
 
 def format_time_lambda(
     exp_class: t.Type[E], dialect: str, default: t.Optional[bool | str] = None
-) -> t.Callable[[t.Sequence], E]:
+) -> t.Callable[[t.List], E]:
```

Comment:
why are you changing this one?

Reply:
Mypy was yelling because e.g. in postgres we have:

```python
args[2] = exp.to_interval(step.this)
```

Then, I checked and in all the places where we have `some_func(args: t.Sequence)`, we have `some_func` in the `FUNCTIONS` map of the parser. This is only used in `_parse_function` and we're always passing in a _list_ of args that's returned from `_parse_csv`, so `t.Sequence` was more general than needed in this case and decided to refactor them all in one go for consistency.